uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -333,6 +333,16 @@ define([
           }
 
           return $q.resolve();
+        },
+
+        /**
+         * Checks if user can upload more file, it totals the number of already
+         * uploaded files and those which are in queue and compares it to limit.
+         *
+         * @return {Boolean} true is user can upload more else false
+         */
+        canUploadMore: function () {
```

Comment:
I don't think this logic should be the responsibility of the instance, rather it should be the controller's. All that the instance knows and should know is that it can own some attachments, but it's the whole app that , knowing what the limit is on the max allowed files (in the settings module), should enforce said limit